### DISCUSSION TITLE
Soft links for the "uri" value

### DISCUSSION THREAD

**Discussion by tao-bioinfo** [GMOD/jbrowse-components#1580] [General] [answered]
I set up a jbrowse instance and the "volvox" test data runs well. Then I make the following modifications to replace `volvox.2bit` with a soft link:

```
cd test_data/volvox
mv volvox.2bit /some/path/
ln -s /some/path/volvox.2bit volvox.2bit
```

Then JBrowse failed to launch as it complaints no `volvox.2bit` file. When I replace `volvox.2bit` with the original file, the session became normal again.

Since my real genome file is very large, I hope to put it in other directories outside the JBrowse path. Possible to do that?

**Comment by cmdcolin** [comment, marked as answer]
This should actually be fine to do this. What server are you using? Maybe the server cannot traverse symlinks?

**Reply by tao-bioinfo** [reply]
I use `npx` introduced in https://jbrowse.org/jb2/docs/quickstart_web

**Reply by cmdcolin** [reply]
It seems like probably the `serve` package server (`npx serve` runs the npm `serve` package https://www.npmjs.com/package/serve) does not support symlinks by default. It requires a custom configuration e.g. like this https://github.com/vercel/serve-handler#options


If you use a different server like apache or nginx then you will have normal symlink support out of the box. I'd probably just use the serve package for quick testing and switch to a normal web server if possible and then you can use symlinks

**Reply by tao-bioinfo** [reply]
Thank you for your suggestion. However I can't find a tutorial on how to use apache for JBrowse 2 in https://jbrowse.org/jb2/docs/.

Besides, I've noticed that some features in JBrowse 1 haven't been imported in JBrowse 2 yet. (Such as Name Searching https://github.com/GMOD/jbrowse-components/discussions/1484) which is very important to me. I've turned to JBrowse 1 temporarily.

I hope to list the features that exist in JBrowse 1 but haven't yet been implemented in JBrowse 2 in https://jbrowse.org/jb2/features#feature-comparison-of-jbrowse-2-vs-jbrowse-1 .

**Reply by cmdcolin** [reply]
Thanks, I started a couple things here https://github.com/GMOD/jbrowse-components/pull/1581. It's true that there are some missing features in jbrowse 2 

We will try to get those things added as soon as possible:)

**Comment by rbuels** [comment]
Yes, JB2 can point to data at any URL. In fact, I think it's best (though
there are diverse opinions on this within the JB2 development team) to keep
your data outside of the JB2 directory and just use absolute URIs in the
configuration to point your JB2 at it.

**Reply by tao-bioinfo** [reply]
OK. But how can I use powerful http servers such as apache for JB2? I can't find documents in https://jbrowse.org/jb2/docs/

**Reply by cmdcolin** [reply]
@billzt the basic idea is that jbrowse 2 (like jbrowse 1) is "static site compatible" meaning that you can unzip the release for jbrowse 2 into /var/www/html/jbrowse2/ folder for example, and then visit http://localhost/jbrowse2/

.... could use a little better documentation though

**Reply by tao-bioinfo** [reply]
OK. So easy? I was mislead by this description in https://jbrowse.org/jb2/docs/quickstart_web:

```
JBrowse 2 requires a web server to run. It won't work if you try to directly open the index.html in your web browser.
```

Then I had thought it was speicial like a Flask app or ruby app that needs special configuration to Apache.

**Reply by cmdcolin** [reply]
Ah, thanks for pointing out this sentence as being potentially confusing. That sentence about "directly opening the index.html" was referring to double-clicking the index.html and opening up the file:/// URL version...we will try to clarfiy that.

JBrowse 2 is very much the same style as JBrowse 1 and so it should work with your existing style of JBrowse 1 server setup
